### PR TITLE
fix generate-key as binary

### LINKED ISSUE #134: `.bin` is not registered when using yarn
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the regression has not already been reported

### Last working version

4.1.0

### Stopped working in version

4.1.1

### Node.js version

18.x

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

20.04

### 💥 Regression Report

`yarn` do not put the `bin` file inside `.bin` folder.
So, it is unavailable to use either `yarn` or `npx` to execute `genkey.js` when you use `yarn` as package manager.

I don't know if it is expected behavior for `yarn`.
Some related information is that `yarn` only consider `/^(?!\.{0,2}$)[a-z0-9._-]+$/i` as valid `bin`.

### Steps to Reproduce

```js
npm init
yarn add @fastify/secure-session
yarn @fastify/secure-session // fail
npx @fastify/secure-session // fail
```

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #134 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Let's create a separate package. Unfortunately we can't rely on node_modules being there.

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "types": "index.d.ts",
   "bin": {
-    "@fastify/secure-session": "genkey.js"
+    "secure-session-gen-key": "./bin/generate-key.js"
```
Comment (climba03003) [inline package.json:8]: It is not resolving the issue. The ultimate goal is using `npx @fastify/secure-session`

**Review by climba03003 — CHANGES_REQUESTED** [review]

**Comment by Uzlopak** [comment]
@mcollina 
@climba03003 

This is my proposal to solve the issue. As long yarn and npx does not support '/ as bin operations we could do this. When they someday implemented it on their end, we could implement it without an additional package.

File: package.json (hunk 1 above)
Comment (Uzlopak) [reply package.json]: tbh I dont want to handle the issue in the npm/yarn codebase.

**Comment by climba03003** [comment]
> This is my proposal to solve the issue. As long yarn and npx does not support '/ as bin operations we could do this. When they someday implemented it on their end, we could implement it without an additional package.

The problem comes from `yarn` only, not `npx`. When ever it use `yarn` to install the package, it fail to link the `bin`.

**Pull request closed without merging**